.github/workflows/go-build-test.yml

```diff
@@ -15,30 +15,102 @@ on:
       - 'go.sum'
 
 jobs:
-  build-and-test:
+  build-and-test-multi-arch:
     runs-on: ubuntu-latest
+    name: Build on ${{ matrix.arch }}
+    # Run steps on a matrix of 2 arch.
+    strategy:
+      matrix:
+        arch:
+          - x64
+          - aarch64
     steps:
-      - uses: actions/checkout@v3
-
-      - name: Set up Go
-        uses: actions/setup-go@v3
-        with:
-          go-version: '1.18'
-          check-latest: true
-
-      - name: Build UI
-        run: make ui/build
-
-      - name: Test
-        run: make go/test
-
-      - name: Build
-        run: make build
-
-      - name: Archive generatated artifacts
-        uses: actions/upload-artifact@v3
-        with:
-          name: parca-bin
-          if-no-files-found: error
-          path: |
-            bin
+    - uses: actions/checkout@v3
+    - name: Set up Go
+      uses: actions/setup-go@v3
+      with:
+        go-version: '1.18'
+        check-latest: true
+
+    - name: Build UI
+      if: ${{ matrix.arch == 'x64' }}
+      run: |
+        make ui/build
+
+    - name: Test
+      if: ${{ matrix.arch == 'x64' }}
+      run: |
+        make go/test
+
+    - name: Build
+      if: ${{ matrix.arch == 'x64' }}
+      run: |
+        make build
+
+    - name: Archive generatated artifacts
+      if: ${{ matrix.arch == 'x64' }}
+      uses: actions/upload-artifact@v3
+      with:
+        name: parca-bin
+        if-no-files-found: error
+        path: |
+          bin
+
+    - name: 'Build and Test on ${{ matrix.arch }}'
```

Comment:
I think it would be more readable to split this into multiple steps.

Reply:
That makes sense, different steps can be. 

- installing necessary packages and golang
- installing and setting up node and yarn
- Building ui
- Running test
- Running build

Do you agree?

Reply:
Sounds good) :+1: